File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;
class Solution {
    public int[] solution(String[] operations) {   
        TreeSet<Integer> ts = new TreeSet<>();
        for (String opStr : operations) {
        StringTokenizer st = new StringTokenizer(opStr);
        if (st.nextToken().equals("I")) {
            int value = Integer.parseInt(st.nextToken());
            ts.add(value);
            } else if (!ts.isEmpty() && opStr.equals("D 1")) {
                ts.pollLast();
            } else if (!ts.isEmpty() && opStr.equals("D -1")) {
                ts.pollFirst();
            }
        }
        return ts.isEmpty() ? new int[] {0, 0} : new int[] {ts.last(), ts.first()};

    }
}

File: 프로그래머스/Lv.3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
const ts = new Set();

        for (const opStr of operations) {
            const st = opStr.split(" ");

            if (st[0] === "I") {
                const value = parseInt(st[1]);
                ts.add(value);
            } else if (ts.size > 0 && opStr === "D 1") {
                ts.delete(Math.max(...Array.from(ts.values())));
            } else if (ts.size > 0 && opStr === "D -1") {
                ts.delete(Math.min(...Array.from(ts.values())));
            }
        }

        return ts.size === 0 ? [0, 0] : [Math.max(...Array.from(ts.values())), Math.min(...Array.from(ts.values()))];
}